src/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/SignedCms.cs

```diff
@@ -198,7 +269,12 @@ public void ComputeSignature(CmsSigner signer, bool silent)
                 // the copy of _heldContent or _contentType here if we're attached.
                 if (!Detached)
                 {
-                    _signedData.EncapContentInfo.Content = content;
+                    using (AsnWriter writer = new AsnWriter(AsnEncodingRules.DER))
```

Comment:
Is this always writing empty octet string? (_hasData == false) if so we could cache this value

Reply:
!_hasData here means "we're the first signer" and thus "we have to build the 'document data' in addition to the 'signer data'".  It doesn't mean that content is empty.

Reply:
I presume double wrapping (contentType != Pks7Data) is intended here?

Reply:
Well, it's single wrapping.  This is putting back the OCTET STRING that got lost by changing from `[OctetString]` to `[AnyValue]` in the serializer type.  The current CMS RFCs say to always write the OCTET STRING.

Rather than match what NetFx writes I just went with "be capable of reading what NetFx can", but since the CMS update was from 1999, it seems that the "new" format should be pretty well supported by readers.